Bottom: 5e38da2a083d36adad2bfbb3e631d2bdd9ce82a1
Top:    4b5b5d846b8b84f1b7037b14b0538d26d69c192b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-04 17:52:58 +0000

Fix z-coord when flipping the board


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index feed8b7..5065ac0 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1800,7 +1800,8 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
 
   glPushMatrix ();
   glScalef ((ghid_flip_x ? -1. : 1.) / gport->zoom,
-            (ghid_flip_y ? -1. : 1.) / gport->zoom, 1);
+            (ghid_flip_y ? -1. : 1.) / gport->zoom,
+            (ghid_flip_x == ghid_flip_y) ? 1. : -1.);
   glTranslatef (ghid_flip_x ? gport->view_x0 - PCB->MaxWidth  :
                              -gport->view_x0,
                 ghid_flip_y ? gport->view_y0 - PCB->MaxHeight :
